pkg/build/builder/docker.go

```diff
@@ -103,7 +106,9 @@ func (d *DockerBuilder) Build() error {
 			if err != docker.ErrNoSuchImage {
 				return err
 			}
-			imageExists = false
+			if !stageNames.Has(imageName) {
+				imageExists = false
+			}
```

Comment:
might as well do this check *before* calling InspectImage, right?  No point in doing the InspectImage call if you don't care about the result.